components/Templates/Neutrum/index.js

```diff
@@ -94,6 +94,22 @@ const schema = {
             <NarrowContainer {...attributes}>{children}</NarrowContainer>,
           editorModule: 'center',
           rules: [
+            {
+              matchMdast: matchZone(
+                'EMBED'
+              ),
+              component: props => (
+                <pre
+                  style={{lineHeight: '1em'}}
+                >
+                  {JSON.stringify(props.embed, null, 3)}
+                </pre>
+              ),
+              editorModule: 'embed',
+              editorOptions: {
+                lookupType: 'paragraph'
```

Comment:
it would be nice to configure url regex here, twitter will require a completely different render component than responsive video – i'd want to use two instances of the embed module, one for responsive video embeds (with thumbnail, click to play and prefetched aspect ratio if possible) and one for twitter

Reply:
Injecting regex implies that the expressions can be customized while that clearly is not the case. The backend has to explicitly analyze and accept the url's anyway. While you can parameterize the regex, it still will or won't get accepted by the backend no matter how you write your expressions.

As for the renderers: I get how a module config for each embed type could help with the render output, but, IMHO, that is premature and will just add further overhead to the schema layer while not adding any real value. You can already perfectly branch to different renderers using the __typename. The only real advantage for further modularization is the flexibility it would offer in terms of what embeds should be available in a template. Again, to me, that seems pretty premature as well.

Reply:
I could live with explicitly named modules as in:
```
{
  matchMdast: matchZone(
    'TWITTER_EMBED'
  ),
  component: CustomTweetComponent,
  editorModule: 'twitterEmbed',
  editorOptions: {
    lookupType: 'paragraph'
  }
},
{
  matchMdast: matchZone(
    'YOUTUBE_EMBED'
  ),
  component: CustomYoutubeComponent,
  editorModule: 'youtubeEmbed',
  editorOptions: {
    lookupType: 'paragraph'
  }
},
{
  matchMdast: matchZone(
    'VIMEO_EMBED'
  ),
  component: CustomVimeoComponent,
  editorModule: 'vimeoEmbed',
  editorOptions: {
    lookupType: 'paragraph'
  }
}
```
They'd all live in and be exposed by the embeds module, where you can freely compose expressions and stuff. 
One con to this approach would be, that we'd add one `onKeyDown` listener per embed. This may have performance implications once the number of `onKeyDown` handlers grows (which it most likely will).

Reply:
I'm fine with explicitly named modules as well but let's go with `embedTweet` and `embedVideo`. `embed` first like `listItem`. Vimeo and YouTube together because they have the same functionality. Eventually will add support for our private videos on Vimeo with our custom player but we can't decide that based on the url (https://vimeo.com/214306781 is ours, https://vimeo.com/242527960 isn't) – in that case the backend will end up making that decision by either including a `mp4`, `hls` and `subtitle` url or not.

Reply:
Imo, this naming suggests a verb and therefore an action, but alright.

> in that case the backend will end up making that decision by either including a mp4, hls and subtitle url or not.
I think, the backend should return a different type entirely when detecting a private or Republik-owned video. But that's cosmetics really.